app/javascript/containers/DatasetsPage.js

```diff
@@ -1,15 +1,121 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import { Table, DatasetModal, Modal, Notification } from 'components';
+
 import AdminContainer from 'containers/shared/AdminContainer';
-import Datasets from 'pages/admin/datasets';
 
 class DatasetsPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedRow: null,
+      deleteWarning: false,
+      deleteSuccess: false,
+      deleteError: false
+    };
+  }
+
+  onClickAction(action, row) {
+    if (action === 'delete') {
+      this.setState({ deleteWarning: true, selectedRow: row });
+    } else if (action == 'toggle') {
+      fetch(window.location.origin + row.enable.value, {
+        method: 'PUT',
+        credentials: 'same-origin'
+      }).then(() => {
+        window.location.reload();
+      });
+    } else {
+      window.location = row.edit.value;
+    }
+  }
+
+  onClickDelete() {
+    const { selectedRow } = this.state;
+    fetch(selectedRow.delete.value, {
+      method: 'DELETE',
+      credentials: 'include'
+    }).then(() => {
+      window.location.reload();
+    }).catch(() => {
+      this.setState({
+        deleteSuccess: false,
+        deleteError: true,
+        deleteWarning: false
+      });
+    });
+  }
+
   render() {
+    const { datasets } = this.props;
+    const { deleteWarning, deleteSuccess, deleteError, selectedRow } = this.state;
+
+    const rows = datasets.map(dataset => {
+      dataset.created = { value: (new Date(dataset.created.value)).toLocaleString(), sortable: true };
+      dataset.edited = { value: (new Date(dataset.edited.value)).toLocaleString(), sortable: true };
+      return dataset;
+    });
+
     return (
-      <AdminContainer>
-        <Datasets datasets={this.props.datasets} />
-      </AdminContainer>
+      <div className="l-page-list">
+        {deleteWarning && (
+          <Modal
+            show={deleteWarning}
+            onClose={() => this.setState({ deleteWarning: false })}
+          >
+            <div className="details-modal">
+              <h1>Are you sure you want to delete this dataset?</h1>
+              {selectedRow.widgets.value && (
+                <div>
+                  <p>The deletion can't be reversed and it will affect to the following widgets: </p>
+                  <p>{selectedRow.widgets.value}</p>
+                </div>
+              )}
+            </div>
+            <footer>
+              <button type="button" className="c-button -outline -dark-text" onClick={() => this.setState({ deleteWarning: false })}>
+                Cancel
+              </button>
+              <button type="button" className="c-button" onClick={() => this.onClickDelete()}>
```

Comment:
Would you be able to put some space between the two buttons? You'll need to add a tiny bit of CSS.

I think the best solution would be to put all of the content of the modal inside a div: `<div className="c-delete-dataset-modal" />` (the class name can be anything here). And then you need to create a SCSS file in `app/assets/stylesheets/components/management/`, named the same, so for example `c-delete-dataset-modal.scss`.

In the file, you could put something like this:
```scss
.c-dataset-modal {
  footer {
    button + button {
      margin-left: 20px;
    }
  }
}
```